jax_md/elasticity.py

```diff
@@ -0,0 +1,526 @@
+      with a standard space will satisfy this property. 
+    tether_strength: scalar. Strength of the "tether" applied to each particle, 
+      which can be necessary to make the Hessian matrix non-singular. Solving
+      for the non-affine response of each particle requires that the Hessian is
+      positive definite. However, there can often be zero modes (eigenvectors of
+      the Hessian with zero eigenvalue) that do not couple to the boundary, and 
+      therefore do not affect the elastic constants despite the zero eigenvalue.
+      The most common example is the global translational modes. To solve for
+      the non-affine response, we consider the "tethered Hessian"
+        H + tether_strength * I, 
+      where I is the identity matrix and tether_strength is a small constant.
+    gradient_check: None or scalar. If not None, a check will be performed 
+      to guarantee that the maximum component of the gradient is less than
+      gradient_check. In other words, that
+        jnp.amax(jnp.abs(grad(energy_fn)(R, box=box))) < gradient_check == True
+      NOTE: JAX currently does not support proper runtime error handling.
+      Therefore, if this check fails, the calculation will return an array
+      of jnp.nan's. It is the users responsibility, if they want to use this
+      check, to then ensure that the returned array is not full of nans.
+    cg_tol: scalar. Tolorance used when solving for the non-affine response.
+    check_convergence: bool. If true, calculate_EMT will return a boolean
+      flag specifiying if the cg solve routine converged to the desired
+      tolorance. The default is False, but convergence checking is highly
+      recommended especially when using 32-bit precision data.
+  
+  Return: A function to calculate the elastic modulus tensor
+
+  """
+
+  def calculate_EMT(R: Array,
+                    box: Array, 
+                    **kwargs) -> Array:
+    """Calculate the elastic modulus tensor
+
+    energy_fn(R) corresponds to the state around which we are expanding
+      
+    Args:
+      R: array of shape (N,dimension) of particle positions. This does not
+        generalize to arbitrary dimensions and is only implemented for
+          dimension == 2
+          dimension == 3
+      box: A box specifying the shape of the simulation volume. Used to infer the
+      volume of the unit cell.
+    
+    Return: C or the tuple (C,converged)
+      where C is the Elastic modulus tensor as an array of shape (dimension,
+      dimension,dimension,dimension) that respects the major and minor 
+      symmetries, and converged is a boolean flag (see above).
+
+    """
+    if not (R.shape[-1] == 2 or R.shape[-1] == 3):
+      raise AssertionError('Only implemented for 2d and 3d systems.')
+
+    if R.dtype is not jnp.dtype('float64'):
+      print("WARNING: elastic modulus calculations can sometimes loose precision when not using 64-bit precision.")
```

Comment:
Here and throughout, can you make sure that all lines are at most 80 characters long?

Also, can this be a `logging.warning` rather than a print?

Also also, loose -> lose?